awssql/plugins/aurora_initial_connection_strategy_plugin.go

```
@@ -0,0 +1,421 @@
+				// and return the current (writer) connection.
+				if isInitialConnection {
+					plugin.hostListProviderService.SetInitialConnectionHostInfo(readerCandidate)
+				}
+				return readerCandidateConn, nil
+			}
+			_ = readerCandidateConn.Close()
+			time.Sleep(time.Millisecond * time.Duration(retryDelayMs))
+			continue
+		}
+		if isInitialConnection {
+			plugin.hostListProviderService.SetInitialConnectionHostInfo(readerCandidate)
+		}
+		return readerCandidateConn, nil
+	}
+	return nil, nil
+}
+
+func (plugin *AuroraInitialConnectionStrategyPlugin) handleGetVerifiedRoleConnectionError(
+	candidateConn driver.Conn,
+	candidate *host_info_util.HostInfo,
+	err error) bool {
+	if err != nil {
+		if candidateConn != nil {
+			_ = candidateConn.Close()
+		}
+		if plugin.pluginService.IsLoginError(err) {
+			return false
+		} else {
+			if candidate != nil {
+				plugin.pluginService.SetAvailability(candidate.GetAllAliases(), host_info_util.UNAVAILABLE)
+			}
+			return true
+		}
+	}
+	return true
+}
+
+func (plugin *AuroraInitialConnectionStrategyPlugin) getReader(props *utils.RWMap[string, string], awsRegion string) (*host_info_util.HostInfo, error) {
+	strategy := property_util.GetVerifiedWrapperPropertyValue[string](props, property_util.READER_HOST_SELECTOR_STRATEGY)
+	if plugin.pluginService.AcceptsStrategy(strategy) {
+		var hostCandidates []*host_info_util.HostInfo
+		if awsRegion != "" {
+			hostCandidates = utils.FilterSlice(
+				plugin.pluginService.GetHosts(),
+				func(hostInfo *host_info_util.HostInfo) bool {
+					return strings.EqualFold(awsRegion, utils.GetRdsRegion(hostInfo.GetHost()))
+				})
+		} else {
+			hostCandidates = plugin.pluginService.GetHosts()
+		}
+
+		host, err := plugin.pluginService.GetHostInfoByStrategy(host_info_util.READER, strategy, hostCandidates)
+		if err != nil {
+			if error_util.IsType(err, error_util.UnsupportedStrategyErrorType) {
+				return nil, err
+			} else {
+				return nil, nil
```

Why are we not returning the err here?

---

Hmm do we think this might be a place to adapt this? Is the err worth logging? And in cases where you know the err has been swallowed lets not call handleGetVerifiedRoleConnectionError